l3_dump.py

```diff
@@ -10,6 +10,12 @@
 import subprocess
 import os
 
+# ##############################################################################
+# Constants that tie the unpacking logic to L3's core structure's layout
+# ##############################################################################
+L3_LOG_HEADER_SZ = 32  # bytes; offsetof(L3_LOG, slots)
+L3_ENTRY_SZ = 32       # bytes; sizeof(L3_ENTRY)
```

Comment:
These constants declare a strong-binding between the layout of the C-struct and what this script expects for log-entry sizes.

There should be a better way to check for this binding, but I could not think of an easy way to do that; i.e. the C file has to export these manifest constants.

I will have to some klunky constants in asserts in the .c file and have this script parse for that.

If you have a really good idea for how-to cross-check these bindings, let me know. Otherwise, I'll sit on it for a while and will come back and improve this in future revs.

Reply:
I have a recollection that there is some utility to do this, but don't remember it. Agree that hard-coding is the way to go for now at least -- simple is good.